### PR TITLE
[Enhancement] Add dead lock checker daemon in FE node

### PR DESCRIPTION
**Pull request by gengjun-git**
Why I'm doing:
Dead lock is a hard problem to solve, we need more information to help find the reason.

What I'm doing:
- Use a map in `QueryableReentrantReadWriteLock` to record the thread who holds the read lock.
- Start a daemon `DeadlockChecker` to check the lock hold time and print the stack if hold time exceeds `Config.slow_lock_threshold_ms`

Fixes #issue

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:

- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.2
  - [x] 3.1
  - [x] 3.0
  - [x] 2.5

### CODE REVIEW

**Review by murphyatwork — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/common/Config.java [hunk 1]
```diff
@@ -814,6 +814,15 @@ public class Config extends ConfigBase {
...
     public static int load_checker_interval_second = 5;
 
+    @ConfField(mutable = true)
+    public static long lock_checker_interval_second = 30;
+
+    /**
+     * Check of deadlock is time consuming. Open it only when tracking problems.
+     */
+    @ConfField(mutable = true)
+    public static boolean enable_deadlock_check = false;
```
Comment (nshangyiming) [inline fe/fe-core/src/main/java/com/starrocks/common/Config.java:824]:
```suggestion
    public static boolean lock_checker_enable_deadlock_detection = false;
```

better to use same prefix for consistent features.

File: fe/fe-core/src/main/java/com/starrocks/consistency/LockChecker.java [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+            LOG.debug("no slow db locks");
+        }
+    }
+
+    private void checkDeadlocks() {
+        if (Config.enable_deadlock_check) {
+            ThreadMXBean tmx = ManagementFactory.getThreadMXBean();
+            long[] ids = tmx.findDeadlockedThreads();
+            if (ids != null) {
+                LOG.info("deadlock threads: {}", ids);
```
Comment (nshangyiming) [inline fe/fe-core/src/main/java/com/starrocks/consistency/LockChecker.java:120]: dump thread stack for deadlock threads?

File: fe/fe-core/src/main/java/com/starrocks/common/Config.java (hunk 1 above)
Comment (gengjun-git) [reply fe/fe-core/src/main/java/com/starrocks/common/Config.java]: done

File: fe/fe-core/src/main/java/com/starrocks/consistency/LockChecker.java (hunk 2 above)
Comment (gengjun-git) [reply fe/fe-core/src/main/java/com/starrocks/consistency/LockChecker.java]: done

**Review by nshangyiming — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.2

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.1

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.0

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.5

**Pull request merged**